coreapps-web-2.2/src/main/java/org/openmrs/module/coreapps/diagnosisCompatibility/DiagnosisServiceCompatibility2_2.java

```diff
@@ -0,0 +1,23 @@
+package org.openmrs.module.coreapps.diagnosisCompatibility;
+
+import java.util.List;
+
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.api.DiagnosisService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+public class DiagnosisServiceCompatibility2_2 implements DiagnosisServiceCompatibility  {
```

Comment:
@mozzy11 but this _could_ be a bean, AFAIC this would even be better, or why not?

@HerbertYiga and where is the pre-2.2 implementation? And more importantly, where is the `DiagnosisServiceCompatibility` API intended to be used?

Reply:
@mks-d  sure , better if it can be made a bean